### PR TITLE
refactor(ci): move crane logic to separate file

### PR DESCRIPTION
**Pull request by dpc**
This should make things more understandable.

### CODE REVIEW

**Review by justinmoon — DISMISSED** [review]

**Review by elsirion — DISMISSED** [review]

**Comment by elsirion** [comment]
Needs rebase :(

File: flake.crane.nix [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+  };
+
+  commonArgsDepsOnly = commonArgsBase // {
+    cargoVendorDir = craneLib.vendorCargoDeps {
+      src = filterWorkspaceFiles ./.;
+    };
+    # copy over the linker/ar wrapper scripts which by default would get
+    # stripped by crane
+    dummySrc = craneLib.mkDummySrc {
+      src = filterWorkspaceDepsBuildFiles ./.;
```
Comment (Maan2003) [inline flake.crane.nix:148]: unrelated to this PR, but

> ./.

This explains why I don't get cached deps (I have cloned it in ~/src/minimint)
https://nix.dev/anti-patterns/language#reproducibility-referencing-top-level-directory-with

**Review by Maan2003 — DISMISSED** [review]

File: flake.crane.nix (hunk 1 above)
Comment (dpc) [reply flake.crane.nix:148]: Oh, you mean from the cachix? Yeah, that would explain it. Will fix.

**Review by Maan2003 — APPROVED** [review]

**Review by jkitman — APPROVED** [review]

File: flake.crane.nix (hunk 1 above)
Comment (dpc) [reply flake.crane.nix:148]: https://github.com/fedimint/fedimint/pull/1934

Comment (elsirion) [reply flake.crane.nix:148]: > This explains why I don't get cached deps (I have cloned it in ~/src/minimint)

Same, thx for the explanation @Maan2003, was always wondering :see_no_evil:

**Pull request merged**